### ISSUE TITLE
Add null_on_oob on all get/gather operations

### ISSUE THREAD

**Issue by orlp** [pola-rs/polars#15240]
Right now only `Expr.list.gather` has the very useful `null_on_oob` option. It allows out-of-bounds indices to return `null`s instead of returning an error. We should expand this to all get/gather operations:

 - `Expr.list.gather` (only function with this option currently)
 - `Expr.list.get` (currently always has the `null_on_oob=True` behavior)
 - ~~`Expr.array.gather`~~ (currently doesn't exist at all)
 - `Expr.array.get` (currently always has the `null_on_oob=True` behavior)
 - `Expr.get` (currently always has the `null_on_oob=False` behavior)
 - `Expr.gather` (currently always has the `null_on_oob=False` behavior)

I think this is fairly uncontroversial, but we do need to decide on a default behavior. Currently `Expr.list.gather`, the only function with the option, has `True` as the default with the following note:

> Note that defaulting to raising an error is much cheaper.

However, I doubt if this would be true in an optimized implementation (maybe a bit cheaper, but I'm doubting the 'much' cheaper).

**Comment by orlp** [comment]
Yes, I did, edited.

**Comment by ritchie46** [comment]
Yes, I think we should add the functionality everywhere and then make the defaults the same on the next breaking release. I think we should default to raising as that's the most strict and informs users they might have a bug in their query.

**Comment by stinodego** [comment]
For 1.0.0, we should at least make sure all these methods have the raise-on-oob behavior by default. We can add parameters to tune the behavior later.
